src/isomorphic/classic/element/ReactElementValidator.js

```diff
@@ -184,7 +184,8 @@ function validatePropTypes(element) {
 var ReactElementValidator = {
 
   createElement: function(type, props, children) {
-    var validType = typeof type === 'string' || typeof type === 'function';
+    var validType = typeof type === 'string' || typeof type === 'function' ||
+                    (type !== null && typeof type === 'object');
```

Comment:
We could add and `Object.keys(type).length` check.

Regardless we probably don't want this part in 15 since it will change the warning behavior (at least before if you had the case you talk about you know something is about to go wrong).